### PR TITLE
Make Resource.Merge consistent with Span.SetAttribute.

### PR DESCRIPTION
**Pull request by Oberon00**
Required for #1346.
Resolves parts of #1184.

As agreed (except for the empty-string change) in the spec meeting, this is a pre-requesite for the updated default resource wording of #1294.

## Changes

- Remove special case for empty string when merging resources.
- Let the second argument take precedence, i.e. x.Merge(y) will have values from y on conflict.

### CODE REVIEW

**Review by anuraaga — APPROVED** [review]
Yes please!

File: specification/resource/sdk.md [hunk 1]
```diff
@@ -43,27 +43,21 @@ Required parameters:
 
 ### Merge
 
-The interface MUST provide a way for a primary resource and a
-secondary resource to be merged into a new resource.
+The interface MUST provide a way for a old resource and a
```
Comment (carlosalberto) [inline specification/resource/sdk.md:46]: *an old

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
LGTM

File: specification/resource/sdk.md (hunk 1 above)
Comment (Oberon00) [reply specification/resource/sdk.md:46]:
```suggestion
The interface MUST provide a way for an old resource and an
```

👍  also *an updating I think.

**Comment by Oberon00** [comment]
Actually, with the new behavior, I would like to change the specified name from `Merge` to `UpdatedWith`. If you agree, please upvote this comment and I can change the PR.

**Comment by anuraaga** [comment]
Preference for Merge over UpdatedWith since it's shorter.

For some reference, JS has the reasonably short `Object.assign` which is similar to the proposed behavior.

**Review by tigrannajaryan — APPROVED** [review]

**Review by yurishkuro — APPROVED** [review]

File: specification/resource/sdk.md [hunk 2]
```diff
@@ -43,27 +43,21 @@ Required parameters:
 
 ### Merge
 
-The interface MUST provide a way for a primary resource and a
-secondary resource to be merged into a new resource.
+The interface MUST provide a way for an old resource and an
```
Comment (jkwatson) [inline specification/resource/sdk.md:46]:
```suggestion
The interface MUST provide a way for an existing resource and an
```

File: specification/resource/sdk.md [hunk 3]
```diff
@@ -43,27 +43,21 @@ Required parameters:
...
 The resulting resource MUST have all attributes that are on any of the two input resources.
-Conflicts (i.e. a key for which attributes exist on both the primary and secondary resource)
-MUST be handled as follows:
-
-* If the value on the primary resource is an empty string, the result has the value of the secondary resource.
-* Otherwise, the value of the primary resource is used.
-
-Attribute key namespacing SHOULD be used to prevent collisions across different
-resource detection steps.
+If a key exists on both the old and updating resource, the value of the updating
```
Comment (jkwatson) [inline specification/resource/sdk.md:54]:
```suggestion
If a key exists on both the existing and updating resource, the value of the updating
```

**Review by jkwatson — APPROVED** [review]

File: specification/resource/sdk.md (hunk 2 above)
Comment (Oberon00) [reply specification/resource/sdk.md:46]: The other resource also exists.

Comment (Oberon00) [reply specification/resource/sdk.md:46]: Maybe "updated", and rename "updating" to "new"?

Comment (Oberon00) [reply specification/resource/sdk.md:46]: Or "previous" and "new".

Comment (jkwatson) [reply specification/resource/sdk.md:46]: "old" makes it sound like it's.... old, which is might not be at all.

Comment (Oberon00) [reply specification/resource/sdk.md:46]: I named it "old" to emphasize that the values in there may be overwritten by the other resource. But I agree that the name is a bit odd. Maybe we could specify it as a member function `resource.Merge(updating)`. Then we don't have to come up with a name for the left-hand-side.

**Review by cijothomas — APPROVED** [review]

File: specification/resource/sdk.md [hunk 4]
```diff
@@ -43,27 +43,21 @@ Required parameters:
...
+resource MUST be picked (even if the updated value is empty).
```
Comment (bogdandrutu) [inline specification/resource/sdk.md:55]: Are we sure about this?

Comment (Oberon00) [reply specification/resource/sdk.md:55]: About the behavior for empty strings? I hope so, it is one special case we lose.

Comment (carlosalberto) [reply specification/resource/sdk.md:55]: Ping @bogdandrutu ;)

**Comment by carlosalberto** [comment]
@Oberon00 Please rebase your branch ;)

**Comment by Oberon00** [comment]
@carlosalberto Done.

**Pull request merged**